homeassistant/components/jvc_projector/config_flow.py

```diff
@@ -0,0 +1,122 @@
+            password = None
+            if CONF_PASSWORD in user_input:
+                password = user_input[CONF_PASSWORD]
+
+            try:
+                if not is_host_valid(host):
+                    raise InvalidHost
+
+                mac = await get_mac_address(host, port, password)
+            except InvalidHost:
+                errors["base"] = "invalid_host"
+            except JvcProjectorConnectError:
+                errors["base"] = "cannot_connect"
+            except JvcProjectorAuthError:
+                errors["base"] = "invalid_auth"
+            else:
+                await self.async_set_unique_id(format_mac(mac))
+                self._abort_if_unique_id_configured(updates={CONF_HOST: host})
+
+                return self.async_create_entry(
+                    title=f"{NAME}",
+                    data={
+                        CONF_HOST: host,
+                        CONF_PORT: port,
+                        CONF_PASSWORD: password,
+                    },
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_HOST): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+                    vol.Optional(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_reauth(self, user_input: Mapping[str, Any]) -> FlowResult:
+        """Perform reauth on password authentication error."""
+        self._reauth_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: Mapping[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        assert self._reauth_entry
+
+        errors = {}
+
+        if user_input is not None:
+            host = self._reauth_entry.data[CONF_HOST]
+            port = self._reauth_entry.data[CONF_PORT]
+            password = user_input[CONF_PASSWORD]
+
+            try:
+                await get_mac_address(host, port, password)
+            except JvcProjectorConnectError:
+                errors["base"] = "cannot_connect"
+            except JvcProjectorAuthError:
+                errors["base"] = "invalid_auth"
+            else:
+                self.hass.config_entries.async_update_entry(
+                    self._reauth_entry,
+                    data={CONF_HOST: host, CONF_PORT: port, CONF_PASSWORD: password},
+                )
+                await self.hass.config_entries.async_reload(self._reauth_entry.entry_id)
+                return self.async_abort(reason="reauth_successful")
+
+        return self.async_show_form(
+            step_id="reauth_confirm",
+            data_schema=vol.Schema({vol.Required(CONF_PASSWORD): str}),
```

Comment:
Why is password a required item in this flow, but optional in the other flow?
How would one set an empty password now?

Reply:
Good catch. Fixed.